src/NServiceBus.Core/Unicast/RunningEndpointInstance.cs

```diff
@@ -61,34 +63,66 @@ public async Task Stop()
 
         public Task Send(object message, SendOptions options)
         {
+            Guard.AgainstNull(nameof(message), message);
+            Guard.AgainstNull(nameof(options), options);
+
+            GuardAgainstUseWhenNotStarted();
             return messageSession.Send(message, options);
         }
 
         public Task Send<T>(Action<T> messageConstructor, SendOptions options)
         {
+            Guard.AgainstNull(nameof(messageConstructor), messageConstructor);
+            Guard.AgainstNull(nameof(options), options);
+
+            GuardAgainstUseWhenNotStarted();
             return messageSession.Send(messageConstructor, options);
         }
 
         public Task Publish(object message, PublishOptions options)
         {
+            Guard.AgainstNull(nameof(message), message);
+            Guard.AgainstNull(nameof(options), options);
+
+            GuardAgainstUseWhenNotStarted();
             return messageSession.Publish(message, options);
         }
 
-        public Task Publish<T>(Action<T> messageConstructor, PublishOptions publishOptions)
+        public Task Publish<T>(Action<T> messageConstructor, PublishOptions options)
         {
-            return messageSession.Publish(messageConstructor, publishOptions);
+            Guard.AgainstNull(nameof(messageConstructor), messageConstructor);
+            Guard.AgainstNull(nameof(options), options);
+
+            GuardAgainstUseWhenNotStarted();
+            return messageSession.Publish(messageConstructor, options);
         }
 
         public Task Subscribe(Type eventType, SubscribeOptions options)
         {
+            Guard.AgainstNull(nameof(eventType), eventType);
+            Guard.AgainstNull(nameof(options), options);
+
+            GuardAgainstUseWhenNotStarted();
             return messageSession.Subscribe(eventType, options);
         }
 
         public Task Unsubscribe(Type eventType, UnsubscribeOptions options)
         {
+            Guard.AgainstNull(nameof(eventType), eventType);
+            Guard.AgainstNull(nameof(options), options);
+
+            GuardAgainstUseWhenNotStarted();
             return messageSession.Unsubscribe(eventType, options);
         }
 
+        void GuardAgainstUseWhenNotStarted()
+        {
+            if (stopping)
```

Comment:
Should this check the `stopped` flag as well? (I see that we never set the stopping flag to false so this would work but might break of we ever set the stopping flag to false once we are stopped)

Reply:
I guess that wouldn't hurt, not needed but is cleaner.